core/src/test/java/org/apache/accumulo/core/metadata/schema/TabletFileTest.java

```diff
@@ -0,0 +1,104 @@
+    assertEquals(TableId.of(tableId), tabletFile.getTableId());
+    assertEquals(tabletDir, tabletFile.getTabletDir());
+    assertEquals(fileName, tabletFile.getFileName());
+  }
+
+  @Test
+  public void testValidPaths() {
+    test("hdfs://localhost:8020/accumulo/tables/2a/default_tablet/F0000070.rf",
+        "hdfs://localhost:8020/accumulo", "2a", "default_tablet", "F0000070.rf");
+    test("hdfs://nn1:9000/accumulo/tables/5a/t-0005/C0009.rf", "hdfs://nn1:9000/accumulo", "5a",
+        "t-0005", "C0009.rf");
+    test(
+        "file:/home/dude/workspace/accumulo/test/target/mini-tests/org.apache.accumulo.test.VolumeIT_test/volumes/v1/tables/1/t-0000003/F0000006.rf",
+        "file:/home/dude/workspace/accumulo/test/target/mini-tests/org.apache.accumulo.test.VolumeIT_test/volumes/v1",
+        "1", "t-0000003", "F0000006.rf");
+  }
+
+  @Test
+  public void testBadPaths() {
+    try {
+      test("C0004.rf", "", "2a", "t-0003", "C0004.rf");
+      fail("Failed to throw error on bad path");
+    } catch (NullPointerException e) {}
+    // 2a< srv:dir
+    try {
+      test("dir", "", "2a", "", "");
+      fail("Failed to throw error on bad path");
+    } catch (NullPointerException e) {}
+    try {
+      test("hdfs://localhost:8020/accumulo/tablets/2a/default_tablet/F0000070.rf",
+          "hdfs://localhost:8020/accumulo", "2a", "default_tablet", "F0000070.rf");
+      fail("Failed to throw error on bad path");
+    } catch (IllegalArgumentException e) {}
+    try {
+      test("hdfs://localhost:8020/accumulo/2a/default_tablet/F0000070.rf",
+          " hdfs://localhost:8020/accumulo", "2a", "default_tablet", " F0000070.rf");
+    } catch (IllegalArgumentException e) {}
+    try {
+      test("/accumulo/tables/2a/default_tablet/F0000070.rf", "", "2a", "default_tablet",
+          "F0000070.rf");
+    } catch (IllegalArgumentException e) {}
+    try {
+      test("hdfs://localhost:8020/accumulo/tables/2a/F0000070.rf", "hdfs://localhost:8020/accumulo",
+          "2a", "", "F0000070.rf");
+    } catch (IllegalArgumentException e) {}
+    try {
+      test("hdfs://localhost:8020/accumulo/tables/F0000070.rf", "hdfs://localhost:8020/accumulo",
+          null, "", "F0000070.rf");
+    } catch (IllegalArgumentException e) {}
+  }
+
+  private final String id = "2a";
+  private final String dir = "t-0003";
+  private final String filename = "C0004.rf";
+
+  @Test
+  public void testFullPathWithVolume() {
+    String volume = "hdfs://1.2.3.4/accumulo";
+    String metadataEntry = volume + "/tables/" + id + "/" + dir + "/" + filename;
+    test(metadataEntry, volume, id, dir, filename);
+  }
+
+  @Test
+  public void testNormalizePath() {
+    String uglyVolume = "hdfs://nn.somewhere.com:86753/accumulo/blah/.././/bad/bad2/../.././/////";
+    String metadataEntry = uglyVolume + "/tables/" + id + "/" + dir + "/" + filename;
+    test(metadataEntry, "hdfs://nn.somewhere.com:86753/accumulo", id, dir, filename);
```

Comment:
Would also be nice to verify that TabletFile.equals and TabletFile.hashCode work as expected for normalized paths.